### ISSUE TITLE
The process of grasping the robotic arm

### ISSUE THREAD

**Issue by ZZ2490** [haritheja-e/robot-utility-models#7]
Hello, how should I carry out the process of grasping the robotic arm? What is the approximate process?

**Comment by haritheja-e** [comment]
Hi! If you’re referring to how we close the gripper that we have on the xArm, we just use a Dynamixel motor which is attached to our gripper mount: https://github.com/haritheja-e/robot-utility-models/tree/main/hardware/xArmMount. After that, we can just actuate the motor to close the gripper [in code](https://github.com/haritheja-e/robot-utility-models/blob/db6f730290d55be763493e5a973e2a7222797af8/robot-server/robot/xarm/xarm.py#L80). One thing to note: our policies are limited to the same gripper used on our data collection tool, but it is pretty low-cost. Please let us know if you have any additional questions!

**Comment by ZZ2490** [comment]
Thank you for your patient reply. Can we use the robotic arm's built-in point cloud camera for scene understanding, instead of attaching an iPhone to the gripper?

**Comment by haritheja-e** [comment]
I'm not sure of the position of the robot arm's built in camera, but if it is similar to that of our iPhone, then it could potentially be used; however, you would need to do an affine transform to match the iPhone POV. We tried this ourselves using the Hello Robot Stretch's default RealSense D405 camera. But, while we were able to get non-zero success for simple tasks (bag and tissue pick up), it failed on other tasks as the focal length of the D405 camera was fixed and did not match the iPhone's focal length that we set (we disable autofocus). I've attached a figure showing images from the two cameras below: 

<img width="1270" alt="Screenshot 2024-12-05 at 10 06 57 PM" src="https://github.com/user-attachments/assets/d0a6b5e5-3025-4d51-86fc-5b4da877d031">

This information can be found in section A.6 of the appendix of our paper. Overall, we found it hard to exactly match the iPhone view with other cameras, and thus could not produce the same success.
